### PR TITLE
Feature hints

### PR DESCRIPTION
**Pull request by LeonWubben**
Implements a hint button to the gui, when clicked it will highlight two pieces randomly that can be swapped with eachother

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java [hunk 1]
```diff
@@ -282,11 +322,14 @@ public void updateCombo(int combo) {
...
     /**
-     * updates the level process bar.
+     * Updates the level progress bar.
+     * progress must be between zero and one, set it to those if it's lower or higher respectively than that
      *
-     * @param progress the percentage of process to show
+     * @param progress the percentage of progress to show
      */
-    public void updateProcessBar(float progress) {
+    private void updateProgressBar(float progress) {
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/view/GUIController.java]: pmd: avoid reassigning parameters

**Comment by albert-smit** [comment]
looks good, when you fix the pmd warning

**Comment by ghyl** [comment]
I hope someone sees that this thing has failing checks

**Comment by LeonWubben** [comment]
Yea I know. Didn't set score for level up back to 2500, fixed that and commit it right after, but forgot to push.
Oops
Done now and checks passed

**Pull request merged**